crates/dpapi/src/client.rs

```diff
@@ -189,10 +190,10 @@ struct GetKeyArgs<'server> {
     negotiate_config: NegotiateConfig,
 }
 
-#[instrument(level = "trace", ret)]
-fn get_key(
+async fn get_key<T: Transport>(
```

Comment:
note: In `sspi`, we used a different approach to avoid asyncifying the API.
We used generators: https://github.com/Devolutions/sspi-rs/blob/3b02a72c5116445df9aa72cfc1d66072dfa71ef8/src/generator.rs

We avoid the need for a fully-fledged async runtime such as Tokio in the native DLL.
I think ideally we may want something like that, but it’s some work to move away from the current asyncified API…

Reply:
Another note: I was working on a library for generators, but I think I was still not satisfied with something. Probably still an improvement over the one we vendored into sspi-rs: https://github.com/CBenoit/genoise

Reply:
> I think ideally we may want something like that, but it’s some work to move away from the current asyncified API…

I figured out the generator's implementation. I can implement it. But I'm not sure if I should do it now. I think I need to ask Mark-André about this task urgency.

> Probably still an improvement over the one we vendored into sspi-rs

Interesting library. I read its code, and I agree, and it looks more cleaner then the `geenerator` module in `sspi-rs`

Reply:
Discussed on Slack, decided to not refactor for now :heavy_check_mark: